build fails on Ubuntu

I did `make cleanall`.
```
Precompile: ─  23.605761 seconds 13.8432%
    JULIA usr/lib/julia/sys-o.a
Illegal inttoptr
          %scevgep9 = ptrtoint i32 addrspace(13)* %scevgep to i64
Illegal inttoptr
          %scevgep1011 = ptrtoint i32 addrspace(13)* %scevgep10 to i64

signal (6): Aborted
in expression starting at no file:0
gsignal at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
abort at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
runOnFunction at /home/sachs/src/julia-master/src/llvm-gc-invariant-verifier.cpp:178
_ZN4llvm13FPPassManager13runOnFunctionERNS_8FunctionE at /home/sachs/src/julia-master/usr/bin/../lib/libLLVM-6.0.so (unknown line)
_ZN4llvm13FPPassManager11runOnModuleERNS_6ModuleE at /home/sachs/src/julia-master/usr/bin/../lib/libLLVM-6.0.so (unknown line)
_ZN4llvm6legacy15PassManagerImpl3runERNS_6ModuleE at /home/sachs/src/julia-master/usr/bin/../lib/libLLVM-6.0.so (unknown line)
operator() at /home/sachs/src/julia-master/src/jitlayers.cpp:1187 [inlined]
jl_dump_native at /home/sachs/src/julia-master/src/jitlayers.cpp:1196
jl_write_compiler_output at /home/sachs/src/julia-master/src/precompile.c:84
jl_atexit_hook at /home/sachs/src/julia-master/src/init.c:233
main at /home/sachs/src/julia-master/ui/repl.c:234
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
_start at /home/sachs/src/julia-master/usr/bin/julia (unknown line)
Allocations: 3781630 (Pool: 3779664; Big: 1966); GC: 9
Aborted (core dumped)
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
make[1]: *** [/home/sachs/src/julia-master/usr/lib/julia/sys-o.a] Error 1
make: *** [julia-sysimg-release] Error 2
```

--- a/deps/llvm.mk
+++ b/deps/llvm.mk
@@ -373,12 +373,14 @@ endif # LLVM_VER
 	touch -c $(LLVM_SRC_DIR)/CMakeLists.txt
 	echo 1 > $@
 
-# Apply version-specific LLVM patches
+# Apply version-specific LLVM patches sequentially
 LLVM_PATCH_PREV :=
 define LLVM_PATCH
 $$(LLVM_SRC_DIR)/$1.patch-applied: $$(LLVM_SRC_DIR)/source-extracted | $$(SRCDIR)/patches/$1.patch $$(LLVM_PATCH_PREV)
 	cd $$(LLVM_SRC_DIR) && patch -p1 < $$(SRCDIR)/patches/$1.patch
 	echo 1 > $$@
+# declare that applying any patch must re-run the compile step
+$$(LLVM_BUILDDIR_withtype)/build-compiled: $$(LLVM_SRC_DIR)/$1.patch-applied
 LLVM_PATCH_PREV := $$(LLVM_SRC_DIR)/$1.patch-applied
 endef
 
@@ -517,7 +519,8 @@ endif
 $(eval $(call LLVM_PATCH,llvm-symver-jlprefix)) # DO NOT REMOVE
 
 
-$(LLVM_BUILDDIR_withtype)/build-configured: $(LLVM_PATCH_PREV)
+# declare that all patches must be applied before running ./configure
+$(LLVM_BUILDDIR_withtype)/build-configured: | $(LLVM_PATCH_PREV)
 
 $(LLVM_BUILDDIR_withtype)/build-configured: $(LLVM_SRC_DIR)/source-extracted | $(llvm_python_workaround) $(LIBCXX_DEPENDENCY)
 	mkdir -p $(dir $@)

